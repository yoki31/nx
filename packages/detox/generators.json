{
  "name": "Nx Detox",
  "version": "0.1",
  "extends": ["@nx/workspace"],
  "generators": {
    "init": {
      "factory": "./src/generators/init/init#detoxInitGeneratorInternal",
      "schema": "./src/generators/init/schema.json",
      "description": "Initialize the `@nx/detox` plugin.",
      "hidden": true
    },
    "application": {
      "factory": "./src/generators/application/application#detoxApplicationGeneratorInternal",
      "schema": "./src/generators/application/schema.json",
      "aliases": ["app"],
      "x-type": "application",
      "description": "Create a Detox application."
    }
  }
}
