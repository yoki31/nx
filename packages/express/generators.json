{
  "name": "Nx Express",
  "version": "0.1",
  "extends": ["@nx/workspace"],
  "generators": {
    "init": {
      "factory": "./src/generators/init/init#initGenerator",
      "schema": "./src/generators/init/schema.json",
      "description": "Initialize the `@nx/express` plugin.",
      "aliases": ["ng-add"],
      "hidden": true
    },
    "application": {
      "factory": "./src/generators/application/application#applicationGeneratorInternal",
      "schema": "./src/generators/application/schema.json",
      "aliases": ["app"],
      "x-type": "application",
      "description": "Create an Express application."
    }
  }
}
