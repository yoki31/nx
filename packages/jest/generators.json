{
  "name": "Nx Jest",
  "version": "0.1",
  "generators": {
    "init": {
      "factory": "./src/generators/init/init#jestInitGeneratorInternal",
      "schema": "./src/generators/init/schema.json",
      "description": "Initialize the `@nx/jest` plugin.",
      "aliases": ["ng-add"],
      "hidden": true
    },
    "configuration": {
      "factory": "./src/generators/configuration/configuration#configurationGeneratorInternal",
      "schema": "./src/generators/configuration/schema.json",
      "description": "Add Jest configuration to a project.",
      "hidden": true
    },
    "convert-to-inferred": {
      "factory": "./src/generators/convert-to-inferred/convert-to-inferred",
      "schema": "./src/generators/convert-to-inferred/schema.json",
      "description": "Convert existing Jest project(s) using `@nx/jest:jest` executor to use `@nx/jest/plugin`."
    }
  }
}
