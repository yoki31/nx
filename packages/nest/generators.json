{
  "name": "nx/nest",
  "version": "0.1",
  "extends": ["@nx/workspace"],
  "generators": {
    "application": {
      "factory": "./src/generators/application/application#applicationGeneratorInternal",
      "schema": "./src/generators/application/schema.json",
      "aliases": ["app"],
      "x-type": "application",
      "description": "Create a NestJS application."
    },
    "init": {
      "factory": "./src/generators/init/init",
      "schema": "./src/generators/init/schema.json",
      "description": "Initialize the `@nx/nest` plugin.",
      "aliases": ["ng-add"],
      "hidden": true
    },
    "library": {
      "factory": "./src/generators/library/library#libraryGeneratorInternal",
      "schema": "./src/generators/library/schema.json",
      "aliases": ["lib"],
      "x-type": "library",
      "description": "Create a new NestJS library."
    },
    "class": {
      "factory": "./src/generators/class/class#classGeneratorInternal",
      "schema": "./src/generators/class/schema.json",
      "description": "Run the `class` NestJS generator with Nx project support."
    },
    "controller": {
      "factory": "./src/generators/controller/controller#controllerGeneratorInternal",
      "schema": "./src/generators/controller/schema.json",
      "description": "Run the `controller` NestJS generator with Nx project support."
    },
    "decorator": {
      "factory": "./src/generators/decorator/decorator#decoratorGeneratorInternal",
      "schema": "./src/generators/decorator/schema.json",
      "description": "Run the `decorator` NestJS generator with Nx project support."
    },
    "filter": {
      "factory": "./src/generators/filter/filter#filterGeneratorInternal",
      "schema": "./src/generators/filter/schema.json",
      "description": "Run the `filter` NestJS generator with Nx project support."
    },
    "gateway": {
      "factory": "./src/generators/gateway/gateway#gatewayGeneratorInternal",
      "schema": "./src/generators/gateway/schema.json",
      "description": "Run the `gateway` NestJS generator with Nx project support."
    },
    "guard": {
      "factory": "./src/generators/guard/guard#guardGeneratorInternal",
      "schema": "./src/generators/guard/schema.json",
      "description": "Run the `guard` NestJS generator with Nx project support."
    },
    "interceptor": {
      "factory": "./src/generators/interceptor/interceptor#interceptorGeneratorInternal",
      "schema": "./src/generators/interceptor/schema.json",
      "description": "Run the `interceptor` NestJS generator with Nx project support."
    },
    "interface": {
      "factory": "./src/generators/interface/interface#interfaceGeneratorInternal",
      "schema": "./src/generators/interface/schema.json",
      "description": "Run the `interface` NestJS generator with Nx project support."
    },
    "middleware": {
      "factory": "./src/generators/middleware/middleware#middlewareGeneratorInternal",
      "schema": "./src/generators/middleware/schema.json",
      "description": "Run the `middleware` NestJS generator with Nx project support."
    },
    "module": {
      "factory": "./src/generators/module/module#moduleGeneratorInternal",
      "schema": "./src/generators/module/schema.json",
      "description": "Run the `module` NestJS generator with Nx project support."
    },
    "pipe": {
      "factory": "./src/generators/pipe/pipe#pipeGeneratorInternal",
      "schema": "./src/generators/pipe/schema.json",
      "description": "Run the `pipe` NestJS generator with Nx project support."
    },
    "provider": {
      "factory": "./src/generators/provider/provider#providerGeneratorInternal",
      "schema": "./src/generators/provider/schema.json",
      "description": "Run the `provider` NestJS generator with Nx project support."
    },
    "resolver": {
      "factory": "./src/generators/resolver/resolver#resolverGeneratorInternal",
      "schema": "./src/generators/resolver/schema.json",
      "description": "Run the `resolver` NestJS generator with Nx project support."
    },
    "resource": {
      "factory": "./src/generators/resource/resource#resourceGeneratorInternal",
      "schema": "./src/generators/resource/schema.json",
      "description": "Run the `resource` NestJS generator with Nx project support."
    },
    "service": {
      "factory": "./src/generators/service/service#serviceGeneratorInternal",
      "schema": "./src/generators/service/schema.json",
      "description": "Run the `service` NestJS generator with Nx project support."
    }
  }
}
