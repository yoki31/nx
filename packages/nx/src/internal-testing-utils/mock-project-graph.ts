jest.doMock('@nx/devkit', () => ({
  ...jest.requireActual<any>('@nx/devkit'),
  createProjectGraphAsync: jest.fn().mockImplementation(async () => {
    return {
      nodes: {},
      dependencies: {},
    };
  }),
}));
