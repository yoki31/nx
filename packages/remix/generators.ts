export { default as actionGenerator } from './src/generators/action/action.impl';
export { default as applicationGenerator } from './src/generators/application/application.impl';
export { default as cypressComponentConfigurationGenerator } from './src/generators/cypress-component-configuration/cypress-component-configuration.impl';
export { default as errorBoundaryGenerator } from './src/generators/error-boundary/error-boundary.impl';
export { default as libraryGenerator } from './src/generators/library/library.impl';
export { default as loaderGenerator } from './src/generators/loader/loader.impl';
export { default as metaGenerator } from './src/generators/meta/meta.impl';
export { default as presetGenerator } from './src/generators/preset/preset.impl';
export { default as resourceRouteGenerator } from './src/generators/resource-route/resource-route.impl';
export { default as routeGenerator } from './src/generators/route/route.impl';
export { default as setupTailwindGenerator } from './src/generators/setup-tailwind/setup-tailwind.impl';
export { default as storybookConfigurationGenerator } from './src/generators/storybook-configuration/storybook-configuration.impl';
export { default as styleGenerator } from './src/generators/style/style.impl';
export { default as initGenerator } from './src/generators/init/init';
