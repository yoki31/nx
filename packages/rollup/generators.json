{
  "name": "Nx rollup",
  "version": "0.1",
  "generators": {
    "init": {
      "factory": "./src/generators/init/init#rollupInitGenerator",
      "schema": "./src/generators/init/schema.json",
      "description": "Initialize the `@nx/rollup` plugin.",
      "aliases": ["ng-add"],
      "hidden": true
    },
    "configuration": {
      "aliases": ["rollup-project"],
      "factory": "./src/generators/configuration/configuration",
      "schema": "./src/generators/configuration/schema.json",
      "description": "Add rollup configuration to a project."
    },
    "convert-to-inferred": {
      "factory": "./src/generators/convert-to-inferred/convert-to-inferred",
      "schema": "./src/generators/convert-to-inferred/schema.json",
      "description": "Convert existing Rollup project(s) using `@nx/rollup:*` executors to use `@nx/rollup/plugin`."
    }
  }
}
