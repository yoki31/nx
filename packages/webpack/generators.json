{
  "name": "Nx Webpack",
  "version": "0.1",
  "generators": {
    "init": {
      "factory": "./src/generators/init/init#webpackInitGeneratorInternal",
      "schema": "./src/generators/init/schema.json",
      "description": "Initialize the `@nx/webpack` plugin.",
      "aliases": ["ng-add"],
      "hidden": true
    },
    "configuration": {
      "aliases": ["webpack-project"],
      "factory": "./src/generators/configuration/configuration#configurationGeneratorInternal",
      "schema": "./src/generators/configuration/schema.json",
      "description": "Add webpack configuration to a project.",
      "hidden": true
    },
    "convert-config-to-webpack-plugin": {
      "factory": "./src/generators/convert-config-to-webpack-plugin/convert-config-to-webpack-plugin",
      "schema": "./src/generators/convert-config-to-webpack-plugin/schema.json",
      "description": "Convert the project to use the `NxAppWebpackPlugin` and `NxReactWebpackPlugin`."
    }
  }
}
