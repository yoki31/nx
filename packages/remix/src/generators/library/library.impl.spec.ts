import 'nx/src/internal-testing-utils/mock-project-graph';

import { readJson, readProjectConfiguration } from '@nx/devkit';
import { type ProjectNameAndRootFormat } from '@nx/devkit/src/generators/project-name-and-root-utils';
import { createTreeWithEmptyWorkspace } from '@nx/devkit/testing';
import applicationGenerator from '../application/application.impl';
import libraryGenerator from './library.impl';

describe('Remix Library Generator', () => {
  describe.each([
    ['derived', 'libs/test'],
    ['as-provided', 'test'],
  ])(
    '-projectNameAndRootFormat=%s',
    (projectNameAndRootFormat: ProjectNameAndRootFormat, libDir) => {
      it('should generate a library correctly', async () => {
        // ARRANGE
        const tree = createTreeWithEmptyWorkspace({ layout: 'apps-libs' });

        // ACT
        await libraryGenerator(tree, {
          name: 'test',
          style: 'css',
          projectNameAndRootFormat,
          addPlugin: true,
        });

        // ASSERT
        const tsconfig = readJson(tree, 'tsconfig.base.json');
        expect(tree.exists(`${libDir}/src/server.ts`));
        expect(tree.children(`${libDir}/src/lib`)).toMatchSnapshot();
        expect(tsconfig.compilerOptions.paths).toMatchSnapshot();
      }, 25_000);

      describe('Standalone Project Repo', () => {
        it('should update the tsconfig paths correctly', async () => {
          // ARRANGE
          const tree = createTreeWithEmptyWorkspace();
          await applicationGenerator(tree, {
            name: 'demo',
            rootProject: true,
            addPlugin: true,
          });
          const originalBaseTsConfig = readJson(tree, 'tsconfig.json');

          // ACT
          await libraryGenerator(tree, {
            name: 'test',
            style: 'css',
            projectNameAndRootFormat,
            addPlugin: true,
          });

          // ASSERT
          const updatedBaseTsConfig = readJson(tree, 'tsconfig.base.json');
          expect(
            Object.keys(originalBaseTsConfig.compilerOptions.paths)
          ).toContain('~/*');
          expect(
            Object.keys(updatedBaseTsConfig.compilerOptions.paths)
          ).toContain('~/*');
        });
      });

      describe('--unitTestRunner', () => {
        it('should not create config files when unitTestRunner=none', async () => {
          // ARRANGE
          const tree = createTreeWithEmptyWorkspace({ layout: 'apps-libs' });

          // ACT
          await libraryGenerator(tree, {
            name: 'test',
            style: 'css',
            unitTestRunner: 'none',
            projectNameAndRootFormat,
            addPlugin: true,
          });

          // ASSERT
          expect(tree.exists(`${libDir}/jest.config.ts`)).toBeFalsy();
          expect(tree.exists(`${libDir}/vite.config.ts`)).toBeFalsy();
        });

        it('should create the correct config files for testing with jest', async () => {
          // ARRANGE
          const tree = createTreeWithEmptyWorkspace({ layout: 'apps-libs' });

          // ACT
          await libraryGenerator(tree, {
            name: 'test',
            style: 'css',
            unitTestRunner: 'jest',
            projectNameAndRootFormat,
            addPlugin: true,
          });

          // ASSERT
          expect(
            tree.read(`${libDir}/jest.config.ts`, 'utf-8')
          ).toMatchSnapshot();
          expect(
            tree.read(`${libDir}/src/test-setup.ts`, 'utf-8')
          ).toMatchSnapshot();
        });

        it('should create the correct config files for testing with vitest', async () => {
          // ARRANGE
          const tree = createTreeWithEmptyWorkspace({ layout: 'apps-libs' });

          // ACT
          await libraryGenerator(tree, {
            name: 'test',
            style: 'css',
            unitTestRunner: 'vitest',
            projectNameAndRootFormat,
            addPlugin: true,
          });

          // ASSERT
          expect(
            tree.read(`${libDir}/vite.config.ts`, 'utf-8')
          ).toMatchSnapshot();

          expect(
            tree.read(`${libDir}/src/test-setup.ts`, 'utf-8')
          ).toMatchSnapshot();
        }, 25_000);
      });

      // TODO(Colum): Unskip this when buildable is investigated correctly
      xit('should generate the config files correctly when the library is buildable', async () => {
        // ARRANGE
        const tree = createTreeWithEmptyWorkspace({ layout: 'apps-libs' });

        // ACT
        await libraryGenerator(tree, {
          name: 'test',
          style: 'css',
          buildable: true,
          projectNameAndRootFormat,
          addPlugin: true,
        });

        // ASSERT
        const project = readProjectConfiguration(tree, 'test');
        const pkgJson = readJson(tree, `${libDir}/package.json`);
        expect(project.targets.build.options.format).toEqual(['cjs']);
        expect(project.targets.build.options.outputPath).toEqual(
          `${libDir}/dist`
        );
        expect(pkgJson.main).toEqual('./dist/index.cjs.js');
        expect(pkgJson.typings).toEqual('./dist/index.d.ts');
      });
    }
  );
});
