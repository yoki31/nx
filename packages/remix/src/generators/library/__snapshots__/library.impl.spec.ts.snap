// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Remix Library Generator -projectNameAndRootFormat=as-provided --unitTestRunner should create the correct config files for testing with jest 1`] = `
"/* eslint-disable */
export default {
  setupFilesAfterEnv: ['./src/test-setup.ts'],
  displayName: 'test',
  preset: '../jest.preset.js',
  transform: {
    '^(?!.*\\\\.(js|jsx|ts|tsx|css|json)$)': '@nx/react/plugins/jest',
    '^.+\\\\.[tj]sx?$': ['babel-jest', { presets: ['@nx/react/babel'] }],
  },
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx'],
  coverageDirectory: '../coverage/test',
};
"
`;

exports[`Remix Library Generator -projectNameAndRootFormat=as-provided --unitTestRunner should create the correct config files for testing with jest 2`] = `
"import { installGlobals } from '@remix-run/node';
import '@testing-library/jest-dom/matchers';
installGlobals();
"
`;

exports[`Remix Library Generator -projectNameAndRootFormat=as-provided --unitTestRunner should create the correct config files for testing with vitest 1`] = `
"import { defineConfig } from 'vite';
import react from '@vitejs/plugin-react';
import { nxViteTsPaths } from '@nx/vite/plugins/nx-tsconfig-paths.plugin';

export default defineConfig({
  root: __dirname,
  cacheDir: '../node_modules/.vite/test',

  plugins: [react(), nxViteTsPaths()],

  // Uncomment this if you are using workers.
  // worker: {
  //  plugins: [ nxViteTsPaths() ],
  // },

  test: {
    setupFiles: ['./src/test-setup.ts'],
    watch: false,
    globals: true,
    cache: { dir: '../node_modules/.vitest/test' },
    environment: 'jsdom',
    include: ['src/**/*.{test,spec}.{js,mjs,cjs,ts,mts,cts,jsx,tsx}'],
    reporters: ['default'],
    coverage: { reportsDirectory: '../coverage/test', provider: 'v8' },
  },
});
"
`;

exports[`Remix Library Generator -projectNameAndRootFormat=as-provided --unitTestRunner should create the correct config files for testing with vitest 2`] = `
"import { installGlobals } from '@remix-run/node';
import '@testing-library/jest-dom/matchers';
installGlobals();
"
`;

exports[`Remix Library Generator -projectNameAndRootFormat=as-provided should generate a library correctly 1`] = `
[
  "test.module.css",
  "test.spec.tsx",
  "test.tsx",
]
`;

exports[`Remix Library Generator -projectNameAndRootFormat=as-provided should generate a library correctly 2`] = `
{
  "@proj/test": [
    "test/src/index.ts",
  ],
  "@proj/test/server": [
    "test/src/server.ts",
  ],
}
`;

exports[`Remix Library Generator -projectNameAndRootFormat=derived --unitTestRunner should create the correct config files for testing with jest 1`] = `
"/* eslint-disable */
export default {
  setupFilesAfterEnv: ['./src/test-setup.ts'],
  displayName: 'test',
  preset: '../../jest.preset.js',
  transform: {
    '^(?!.*\\\\.(js|jsx|ts|tsx|css|json)$)': '@nx/react/plugins/jest',
    '^.+\\\\.[tj]sx?$': ['babel-jest', { presets: ['@nx/react/babel'] }],
  },
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx'],
  coverageDirectory: '../../coverage/libs/test',
};
"
`;

exports[`Remix Library Generator -projectNameAndRootFormat=derived --unitTestRunner should create the correct config files for testing with jest 2`] = `
"import { installGlobals } from '@remix-run/node';
import '@testing-library/jest-dom/matchers';
installGlobals();
"
`;

exports[`Remix Library Generator -projectNameAndRootFormat=derived --unitTestRunner should create the correct config files for testing with vitest 1`] = `
"import { defineConfig } from 'vite';
import react from '@vitejs/plugin-react';
import { nxViteTsPaths } from '@nx/vite/plugins/nx-tsconfig-paths.plugin';

export default defineConfig({
  root: __dirname,
  cacheDir: '../../node_modules/.vite/libs/test',

  plugins: [react(), nxViteTsPaths()],

  // Uncomment this if you are using workers.
  // worker: {
  //  plugins: [ nxViteTsPaths() ],
  // },

  test: {
    setupFiles: ['./src/test-setup.ts'],
    watch: false,
    globals: true,
    cache: { dir: '../../node_modules/.vitest/libs/test' },
    environment: 'jsdom',
    include: ['src/**/*.{test,spec}.{js,mjs,cjs,ts,mts,cts,jsx,tsx}'],
    reporters: ['default'],
    coverage: { reportsDirectory: '../../coverage/libs/test', provider: 'v8' },
  },
});
"
`;

exports[`Remix Library Generator -projectNameAndRootFormat=derived --unitTestRunner should create the correct config files for testing with vitest 2`] = `
"import { installGlobals } from '@remix-run/node';
import '@testing-library/jest-dom/matchers';
installGlobals();
"
`;

exports[`Remix Library Generator -projectNameAndRootFormat=derived should generate a library correctly 1`] = `
[
  "test.module.css",
  "test.spec.tsx",
  "test.tsx",
]
`;

exports[`Remix Library Generator -projectNameAndRootFormat=derived should generate a library correctly 2`] = `
{
  "@proj/libs/test": [
    "libs/test/src/index.ts",
  ],
  "@proj/libs/test/server": [
    "libs/test/src/server.ts",
  ],
}
`;
